src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -67,24 +67,29 @@ public override void OnStartActivity(Activity activity, object payload)
             var request = context.Request;
             if (!this.hostingSupportsW3C || !(this.options.TextFormat is TraceContextFormat))
             {
-                // This requires to ignore the current activity and create a new one
-                // using the context extracted from w3ctraceparent header or
-                // using the format TextFormat supports.
-
                 var ctx = this.options.TextFormat.Extract(default, request, HttpRequestHeaderValuesGetter);
-                if (ctx != default)
+

```

Comment:
@cijothomas @reyang Pay close attention to this part. Here is where it gets dicey.

* aspnetcore doesn't parse "Baggage" it only look for "Correlation-Context" (I'm [mostly sure](https://github.com/dotnet/aspnetcore/blob/7e139c9b5f0dc4a2f6da5f413092f10365a4d644/src/Http/Headers/src/HeaderNames.cs#L39) about that). What that means is we have to extract now even on .NET Core. We should work with the .NET team to add that support as soon as the spec is ~done so we can optimize that away?

* There was a bug here before. If anyone was using something other than TraceContextFormat, we always created a new Activity. Even if you were doing `CompositePropagator(TraceContextFormat)`! I fixed it so we only create a new Activity if the parsed context doesn't match.

* Baggage will be applied to the final activity.

Reply:
Yeah, fully aware of this.
Given the W3C Baggage spec is in its very early stage, so it could be renamed again (and it has been renamed 3 times IIRC).

Reply:
seeing this comment late..
Copying my comment here:

ASP.NET (and ASP.NET Core) already extracts W3CTraceParent and creates ActivityContext. We'll now be forced to extract it once more time by default, and compare it with the one extracted by ASP.NET. Considering this is in hot path, before sampling, won't we regress perf ?

Reply:
Ya, that's what I'm saying above. We have to double-extract, which will cost perf. It's a tradeoff. Perf for support of Baggage out of the box. Not sure if the benefit is worth the perf hit, that's what we need to discuss. I did just make a change so ITextFormat implementations won't double-extract. That will lessen the perf hit.